### PR TITLE
[feat] create simple interest form component

### PR DESCRIPTION
**Pull request by varortz**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

## :tanabata_tree: Description
- Converts the interest page into a component, and adds additional form inputs necessary (radio inputs & text input for start date)
- Implements a Button component in `components/Button.tsx`
- Passes in CaseListing objects into both the CaseDetails and InterestForm component to facilitate displaying/using case information for a specific case 

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @davidqing6432 at the bottom."

iForm

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
- Fetch this branch and go on it
- navigate to the cases page in the url: `http://localhost:3000/cases`
- scroll through the cases and click on random ones, make sure that the interest form looks good throughout all the cases being clicked on
- Click on the radio buttons and use the text inputs. Keep in mind there's currently no validation yet, so just make sure they don't act wonky when being interacted with. 
- Hover over the "Submit Interest" button

Might be good to test all versions of the Button/ErrorButton component
- Look at `components/Button.tsx` for documentation on how to use the buttons 

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Update the CaseDetails component to display case information according to David's designs
- Add validation to the interest form and update the interest submission functionality if necessary (focused more on the frontend aspect in this PR due to schema changes 🙈)
- Fix scrolling on the cases page to facilitate scrolling to see the case details + interest form

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
Guide for creating a custom radio input using styled components (kinda janky implementation, but Jinkang++ helped make edits to it in this PR to improve it):
-  https://medium.com/@christinaroise/how-to-create-a-reusable-custom-radiobutton-in-react-with-typescript-3ae7fc704e09

Styled Components documentation that is useful for understanding the Button component implementation:
https://styled-components.com/docs/basics#adapting-based-on-props

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @pragyakallanagoudar @davidqing6432

### CODE REVIEW

**Review by ccatherinetan — COMMENTED** [review]
Looks rlly good! a couple things to implement: 
- clear out the form responses when clicking on a new case on the scroll bar 
- form validation + updating interests table w/ submission 
- include a warning on the interest form that the form responses are not saved (according to HIFI)
- make the input text border to 2 pixels (David will also change on figma)
-  hover functionality for the Radio Button 

The Button, ErrorButton + their secondary formats look great!

File: src/components/InterestForm/index.tsx
Comment (ccatherinetan) [inline src/components/InterestForm/index.tsx]: delete comment before merging!

File: src/components/InterestForm/styles.tsx
Comment (ccatherinetan) [inline src/components/InterestForm/styles.tsx]: do we want to add this color to COLORS?

File: src/components/Radio.tsx
Comment (ccatherinetan) [inline src/components/Radio.tsx]: are we planning to use the `Radio` component anywhere else? if not, should we delete? otherwise, are we planning to add anything to this styled component?

File: src/app/cases/page.tsx
Comment (ccatherinetan) [inline src/app/cases/page.tsx:136]: is there ever a case where the `caseInfo` is False, and the `CaseDetailDisplay` doesn't show up?

File: src/components/CaseDetails/styles.tsx
Comment (ccatherinetan) [inline src/components/CaseDetails/styles.tsx]: David changed the fonts to Open Sans. I think you want this component to inherit from H1 (a styled h1 defined in `@/styles/text` which has the right size + font + color)

I've updated H1 in my latest sprint, so the font will update after that's merged. However, in the future, if we use styled h1, h2 etc. we should use H1, H2 etc. from `@/styles/text` for consistency 
@davidqing6432 does that sound reasonable?

File: src/components/InterestForm/index.tsx
Comment (ccatherinetan) [inline src/components/InterestForm/index.tsx]: Is there a reason why "What role(s) are you applying for?" is a `<div>`, and the other questions are `<p>`?

I think the text needs different styling (the font color + weight doesn't match the figma). 
Maybe we should create a custom styled P component in a future sprint → use the custom P component for the question texts. 
@davidqing6432 thoughts?

File: src/components/InterestForm/styles.tsx
Comment (ccatherinetan) [inline src/components/InterestForm/styles.tsx]: change to rem?

File: src/components/InterestForm/styles.tsx
Comment (ccatherinetan) [inline src/components/InterestForm/styles.tsx]: change border width to 2px! (here and elsewhere)

File: src/components/Radio.tsx
Comment (ccatherinetan) [inline src/components/Radio.tsx]: 2px border

File: src/components/Button.tsx
Comment (ccatherinetan) [inline src/components/Button.tsx]: do we want to use a color from COLORS, or add this color (`'#f2f2f2'`) to COLORS?

Comment (jinkang-0) [reply src/components/Button.tsx]: oh I added that same color... I called it `lightGray` since that's what David named it on the Figma

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [reply src/components/InterestForm/index.tsx]: Catherine your comments are incredible!! Thanks for your attention to detail :))

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx:48]: think @davidqing6432 may have gotten rid of this? I like it here though tbh

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx:98]: Screenshot 2023-10-28 at 10 50 50 PM

Can we include the little blurb that says "Your interest information is not saved!" etc. as well?

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx:67]: the radio buttons are so beautiful

File: src/components/InterestForm/styles.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/styles.tsx:77]: Screenshot 2023-10-28 at 10 54 45 PM

Can we also include a style for the disabled radio button? i.e. the first one in this image (cc @davidqing6432)

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx:40]: Curious why this function seems to force a re-render of the whole page? Are we explicitly doing this somewhere?

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx:56]: Not sure if this is just on my end, but for me this doesn't seem to be writing an interest to the table? All of this seems fine though, so I'm not sure why this might be 🤔 

But also since this sprint is entirely frontend-focused, we can look deeper into this in your next sprint!

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx:75]: There's a warning about this in the console:

```
Warning: You provided a `checked` prop to a form field without an `onChange` handler. This will render a read-only field. If the field should be mutable use `defaultChecked`. Otherwise, set either `onChange` or `readOnly`.
```

Maybe try to replace `checked` with `defaultChecked` and see if that fixes it?

File: src/components/Button.tsx
Comment (varortz) [reply src/components/Button.tsx]: Updating this! Also, I'll update the COLORS.tsx file so that we don't have more than one variable for the same color. I think it could be good to stick with the names David gave the colors on Figma.

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx]: Thanks for informing me about the 2px border decision for inputs!

File: src/components/InterestForm/index.tsx
Comment (varortz) [reply src/components/InterestForm/index.tsx:75]: changing from onClick to onChange got rid of the console error, thanks for noticing it!

File: src/components/Radio.tsx
Comment (varortz) [reply src/components/Radio.tsx]: I don't think we are, I just forgot to delete this file. Thanks for catching that!

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx]: Fixed, will be pushing new commit soon!

File: src/app/cases/page.tsx
Comment (jinkang-0) [inline src/app/cases/page.tsx:10]: should we refactor to use `@/`?

Comment (jinkang-0) [inline src/app/cases/page.tsx]: I think we should refactor to use `@/` to make things cleaner, though the changes in main branch likely has that (make sure to double check the imports are still right!)

File: src/app/cases/page.tsx
Comment (jinkang-0) [reply src/app/cases/page.tsx:42]: I believe so, when the user hasn't selected a card. Not sure if we want a selected card by default though.

File: src/components/Button.tsx
Comment (jinkang-0) [inline src/components/Button.tsx]: A better name may be `$color`? I don't think `$secondary` clearly states the purpose of the prop

File: src/components/Button.tsx
Comment (jinkang-0) [inline src/components/Button.tsx]: I believe we prefer `rem` units, shall we switch?

File: src/components/CaseDetails/index.tsx
Comment (jinkang-0) [inline src/components/CaseDetails/index.tsx]: Shall we refactor to `@/`?

File: src/components/CaseDetails/styles.tsx
Comment (jinkang-0) [inline src/components/CaseDetails/styles.tsx:19]: Shall we use `rem` here?

File: src/components/InterestForm/index.tsx
Comment (jinkang-0) [inline src/components/InterestForm/index.tsx]: Shall we use `@/` here?

File: src/components/InterestForm/index.tsx
Comment (jinkang-0) [inline src/components/InterestForm/index.tsx]: I believe the Figma has the placeholder as `DD/MM/YYYY`, but is up in the air (cc: @davidqing6432 do we want to follow the North American date format?)

File: src/components/InterestForm/styles.tsx
Comment (jinkang-0) [inline src/components/InterestForm/styles.tsx]: Very dangerous to use `vh` here, I would suggest using `rem` with `min-height` instead. `vh` may seem fine on a desktop view, but once you're in the land of the mobile devices, viewport units get weird.

File: src/components/InterestForm/styles.tsx
Comment (jinkang-0) [inline src/components/InterestForm/styles.tsx:22]: I would suggest adding a `resize: vertical` property to allow the user to resize the `textarea` component. Though this is a change that is dependent on the design. I think it is preferable to improve user experience, especially given that we are already accounting for a vertically overflowing card, but we may want to consult @davidqing6432 ?

File: src/components/InterestForm/styles.tsx
Comment (jinkang-0) [inline src/components/InterestForm/styles.tsx]: I think we want to reference the `COLORS` object instead of using CSS variables, but we should discuss this!

File: src/components/ListingCard/ListingCard.tsx
Comment (jinkang-0) [inline src/components/ListingCard/ListingCard.tsx]: We might want to update to use `@/` here? Or you can move that to the next sprint task!

**Review by jinkang-0 — COMMENTED** [review]
Looks great! I just added some minor details to consider, looking forward to the new changes!

File: src/components/InterestForm/index.tsx
Comment (varortz) [reply src/components/InterestForm/index.tsx]: I messaged @davidqing6432 earlier and he decided on MM/DD/YYYY

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx]: Fixed! Just haven't pushed the changes yet

File: src/app/cases/page.tsx
Comment (jinkang-0) [reply src/app/cases/page.tsx:136]: I think there is the case that initially `caseInfo` is empty, so there will be no case detail to display. Not sure what the consensus is on auto-selecting the first case though, but if we do that then there's the edge case of us having no cases to display.

File: src/components/Button.tsx
Comment (varortz) [reply src/components/Button.tsx]: Named the prop `$secondary` based on the styled-components docs example, the Figma also splits each button type (regular and error) into 'primary' and 'secondary'. I think naming it `$color` can be more confusing in my opinion, but I'm open to changing it. @pragyakallanagoudar thoughts on this??

File: src/app/cases/page.tsx
Comment (varortz) [reply src/app/cases/page.tsx:136]: yup what jinkang said is correct, sorry for not replying to this earlier. When the page first renders, `caseInfo` is undefined as it's set whenever a case card is clicked. This causes an error when trying to pass it in as a prop to the `CaseDetailDisplay` component which is why we need to check that `caseInfo` is not null/undefined. Also with this current implementation, it looks like it defaults to displaying the first case already when the page first renders. I think maybe we should test this page for the edge case where no cases are currently on the backend, and decide on what we would want the behavior to look like?

Comment (pragyakallanagoudar) [reply src/app/cases/page.tsx:136]: i'm a fan of auto-selecting the first case! i think we probably want to discuss this + what the interest form will look like if there are no cases. adding to tmr's meeting agenda, and also cc'ing @davidqing6432 on this thread

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx:22]: yeah, I think this change should be up to David since it messes with the design a bit.

Comment (pragyakallanagoudar) [reply src/components/InterestForm/styles.tsx:22]: yeah we can run by david tmr

File: src/components/Button.tsx
Comment (jinkang-0) [reply src/components/Button.tsx]: ohh I see, sorry I didn't read too much into what `$secondary` does. I still think we should allow for customizations to the background and text colors still (with hover color computed using helpers), which will open up the door for colors other than "secondary" without needing to copy paste this button component. Instead, we can make preset React components that fill in the values for us (i.e. `PrimaryButton`, `SecondaryButton`, `DangerButton` (for the reset password one).

File: src/components/CaseDetails/styles.tsx
Comment (varortz) [reply src/components/CaseDetails/styles.tsx]: updated! 🥳

File: src/components/Button.tsx
Comment (pragyakallanagoudar) [reply src/components/Button.tsx]: ahhh I'm just now understanding this point LOL sorry Jinkang I think I was misinterpreting this at worksession

this is good solution! but again I'll leave it up to you @varortz, i don't have a strong opinion either way

File: src/components/InterestForm/index.tsx
Comment (pragyakallanagoudar) [inline src/components/InterestForm/index.tsx]: can we rename 'Both Attorney & Interpreter' to 'Either Attorney Or Interpreter'? David just made that change today!

File: src/components/Button.tsx
Comment (varortz) [reply src/components/Button.tsx]: Yeah, I already started implementing the button using jinkang's suggestion since it's good practice. Ty @jinkang-0 for the suggestion and clarifications!

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx:22]: I'm going to leave it as a scrollable textarea component for this PR, but I've started looking into making it auto-resizable already! Planning on implementing an auto-resizable version of it soon, and getting the team's (and maybe bestie Ambreen's 🤔) thoughts on what we prefer to move forward with.

File: src/components/InterestForm/index.tsx
Comment (varortz) [reply src/components/InterestForm/index.tsx:56]: Yup, there's a 409 error rn... I'll look more into it during the next sprint to not delay merging the frontend changes in!

File: src/components/InterestForm/index.tsx
Comment (varortz) [reply src/components/InterestForm/index.tsx:40]: I have a rough idea of why it might be re-rendering, but I'll also look more into this during the next sprint! I think once I reconnect the form to the backend it'll be easier to debug.

File: src/components/InterestForm/index.tsx
Comment (varortz) [reply src/components/InterestForm/index.tsx]: thank you frfrf, I should've been adding more things to the global styles to begin with lol 😅 handled it now and will be pushing all the fixes soon!

File: src/styles/text.tsx
Comment (varortz) [inline src/styles/text.tsx:44]: Noticed Catherine added these in her PR and kinda needed them too, so I went ahead and included them here as well. I can deal w rebasing later just lmk!

File: src/components/CaseDetails/styles.tsx
Comment (pragyakallanagoudar) [reply src/components/CaseDetails/styles.tsx:19]: resolved this in slack DMs with sauhard! this is what he said:

> i would rec using px
> reason is rem depends on root font size and border widths have no relation to font size
> and the variance is border width across screen size is pretty limited

Comment (pragyakallanagoudar) [reply src/components/CaseDetails/styles.tsx:19]: so let's keep this as is

File: src/api/supabase/queries/profiles.ts
Comment (pragyakallanagoudar) [inline src/api/supabase/queries/profiles.ts]: @jinkang-0 any idea why this might be here? 🤔 I assumed that this was part of the profile context PR which was already merged in...

File: src/components/Button.tsx
Comment (pragyakallanagoudar) [inline src/components/Button.tsx:3]: AMAZING documentation!! thank you for this alvaro :)))

File: src/api/supabase/queries/profiles.ts
Comment (pragyakallanagoudar) [reply src/api/supabase/queries/profiles.ts]: actually it seems like all the changes from the profile context and global styles are included on here...? i somehow thought this diff would only included changes that were not already present on main after rebasing :o but i might be misremembering!

**Review by pragyakallanagoudar — APPROVED** [review]
LOOKS AMAZINGGGG THANK YOU FOR ALL YOUR HARD WORK ALVARO!!!

File: src/components/CaseDetails/styles.tsx
Comment (jinkang-0) [reply src/components/CaseDetails/styles.tsx:19]: I think for `border-radius` we will still use `rem`, only `px` for `border-width`

File: src/components/Button.tsx
Comment (jinkang-0) [inline src/components/Button.tsx]: We may want to consider using `rem` units here

File: src/components/InterestForm/styles.tsx
Comment (jinkang-0) [inline src/components/InterestForm/styles.tsx]: flagging `px` -> `rem` here also

File: src/components/InterestForm/styles.tsx
Comment (jinkang-0) [inline src/components/InterestForm/styles.tsx]: flagging `px` -> `rem` here also

File: src/components/TextInput/styles.tsx
Comment (jinkang-0) [inline src/components/TextInput/styles.tsx]: flagging `px` -> `rem` here also

File: src/utils/helpers.ts
Comment (jinkang-0) [inline src/utils/helpers.ts]: extra comment?

**Review by jinkang-0 — COMMENTED** [review]
Gonna drop some unit policing stuff

File: src/components/CaseDetails/styles.tsx
Comment (varortz) [reply src/components/CaseDetails/styles.tsx:19]: @pragyakallanagoudar are you fine with going along with this change? I think it makes sense to change `border-radius` to `rem`

File: src/utils/helpers.ts
Comment (varortz) [reply src/utils/helpers.ts]: si, will remove

File: src/components/Button.tsx
Comment (varortz) [reply src/components/Button.tsx]: fixed!

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx]: fixed!

File: src/components/TextInput/styles.tsx
Comment (varortz) [reply src/components/TextInput/styles.tsx]: fixed!

File: src/components/InterestForm/styles.tsx
Comment (varortz) [reply src/components/InterestForm/styles.tsx]: fixed!

**Review by jinkang-0 — APPROVED** [review]

**Pull request merged**
